test/framework/src/main/java/org/elasticsearch/test/AbstractQueryTestCase.java

```diff
@@ -160,25 +160,25 @@ public void testUnknownObjectException() throws IOException {
         // Adds the alternates versions of the query too
         candidates.addAll(getAlternateVersions().keySet());
 
-        List<Tuple<String, Boolean>> testQueries = alterateQueries(candidates, getObjectsHoldingArbitraryContent());
-        for (Tuple<String, Boolean> testQuery : testQueries) {
-            boolean expectedException = testQuery.v2();
+        List<Tuple<String, String>> testQueries = alterateQueries(candidates, getObjectsHoldingArbitraryContent());
+        for (Tuple<String, String> testQuery : testQueries) {
+            String expectedException = testQuery.v2();
             try {
                 parseQuery(testQuery.v1());
-                if (expectedException) {
+                if (expectedException != "") {
```

Comment:
Personally, I think null might be a better choice here. If you want to keep "", I am ok with it, but we should replace `!=` and `==` with `equalTo()`, the only reason this works at the moment is that they both were constants, but I don't think we should rely on that even in test code.

Reply:
I was having problems with null values and I just found why, I was hitting the following JDK bug: https://bugs.openjdk.java.net/browse/JDK-8148463

I agree it might make more sense to use null so I am changing it.